app/routes/login.tsx

```diff
@@ -35,13 +37,38 @@ export const loader: LoaderFunction = async ({ request }) => {
 export const action: ActionFunction = async ({ request }) => {
   const formData = await request.formData();
   const idToken = formData.get("idToken");
+  const accessToken = formData.get("accessToken");
 
   // Save the session in a cookie with the user's tokens
   let session = await getSession(request.headers.get("Cookie"));
   session.set("idToken", idToken);
+  session.set("accessToken", accessToken);
+
+  // Check whether the user has a profile page. If not, ask them to create one.
+  const user = await getCurrentUser(session);
+
+  if (!user) {
+    throw new Error(
+      "Unable to create a profile, please refresh the page and try again"
+    );
+  }
+
+  let redirectPath: string;
+  const profile = await getUserProfileBySlug(user.githubLogin);
```

Comment:
Now that we have a user, we check to see if they have a profile. People who do get sent to their profile page, but for people who are logging in for the first time, we do 2 things:

1. create their user profile in our database
2. send them to their profile with a "Welcome" modal